[2025-06-20T18:36:12.795+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-20T18:36:12.848+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: nba_etl_to_staging.fetch_schedule_and_load_to_stage manual__2025-06-20T18:36:10.215155+00:00 [queued]>
[2025-06-20T18:36:12.861+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: nba_etl_to_staging.fetch_schedule_and_load_to_stage manual__2025-06-20T18:36:10.215155+00:00 [queued]>
[2025-06-20T18:36:12.862+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-06-20T18:36:12.935+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_schedule_and_load_to_stage> on 2025-06-20 18:36:10.215155+00:00
[2025-06-20T18:36:12.945+0000] {standard_task_runner.py:63} INFO - Started process 4291 to run task
[2025-06-20T18:36:12.951+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'nba_etl_to_staging', 'fetch_schedule_and_load_to_stage', 'manual__2025-06-20T18:36:10.215155+00:00', '--job-id', '292', '--raw', '--subdir', 'DAGS_FOLDER/etl_pipeline.py', '--cfg-path', '/tmp/tmphg0mt61x']
[2025-06-20T18:36:12.953+0000] {standard_task_runner.py:91} INFO - Job 292: Subtask fetch_schedule_and_load_to_stage
[2025-06-20T18:36:13.083+0000] {task_command.py:426} INFO - Running <TaskInstance: nba_etl_to_staging.fetch_schedule_and_load_to_stage manual__2025-06-20T18:36:10.215155+00:00 [running]> on host 5ea278fdf3a6
[2025-06-20T18:36:13.261+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='nba_etl_to_staging' AIRFLOW_CTX_TASK_ID='fetch_schedule_and_load_to_stage' AIRFLOW_CTX_EXECUTION_DATE='2025-06-20T18:36:10.215155+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-20T18:36:10.215155+00:00'
[2025-06-20T18:36:13.262+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-20T18:36:13.277+0000] {etl_pipeline.py:131} INFO - Fetching schedule from NBA API...
[2025-06-20T18:36:13.665+0000] {etl_pipeline.py:142} ERROR - Failed to fetch data from NBA API: 403 Client Error: Forbidden for url: https://cdn.nba.com/static/json/staticData/scheduleLeague+V2.json
[2025-06-20T18:36:13.666+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-20T18:36:13.666+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/etl_pipeline.py", line 139, in fetch_and_load_schedule_to_stage
    res.raise_for_status()
  File "/home/airflow/.local/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://cdn.nba.com/static/json/staticData/scheduleLeague+V2.json
[2025-06-20T18:36:13.682+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=nba_etl_to_staging, task_id=fetch_schedule_and_load_to_stage, run_id=manual__2025-06-20T18:36:10.215155+00:00, execution_date=20250620T183610, start_date=20250620T183612, end_date=20250620T183613
[2025-06-20T18:36:13.720+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 292 for task fetch_schedule_and_load_to_stage (403 Client Error: Forbidden for url: https://cdn.nba.com/static/json/staticData/scheduleLeague+V2.json; 4291)
[2025-06-20T18:36:13.744+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-06-20T18:36:13.767+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-20T18:36:13.907+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
